### PR TITLE
chore: scale up max delegators per collator to 35 from 25

### PR DESCRIPTION
**Pull request by wischli**
## fixes KILTProtocol/ticket#1749
* Moves configuration of `MaxDelegatorsPerCollator` to `runtime-common` constant and increases to 35 from 25
* Replaces custom lock extension with Substrate's `extend_lock` which did not exist at the point of initially writing the pallet
* [M](https://kilt-cast.monday.com/boards/1792114080/pulses/2001105863)

## Checklist:

- [x] I have verified that the code works
  - [x] No panics! (checked arithmetic ops, no indexing `array[3]` use `get(3)`, ...)
- [x] I have verified that the code is easy to understand
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

**Review by Diiaablo95 — APPROVED** [review]
LGMT!

**Comment by Diiaablo95** [comment]
@wischli I guess we also need to re-run the benchmarks, right?

**Review by weichweich — APPROVED** [review]
cool

**Comment by wischli** [comment]
> @wischli I guess we also need to re-run the benchmarks, right?

Yes!

File: runtimes/spiritnet/src/weights/parachain_staking.rs [hunk 1]
```diff
@@ -167,29 +167,27 @@ impl<T: frame_system::Config> parachain_staking::WeightInfo for WeightInfo<T> {
...
-			.saturating_add((3_133_000 as Weight).saturating_mul(n as Weight))
-			// Standard Error: 40_000
-			.saturating_add((26_383_000 as Weight).saturating_mul(m as Weight))
-			// Standard Error: 177_000
-			.saturating_add((246_000 as Weight).saturating_mul(u as Weight))
+			// Standard Error: 15_000
+			.saturating_add((4_365_000 as Weight).saturating_mul(n as Weight))
+			// Standard Error: 27_000
+			.saturating_add((27_649_000 as Weight).saturating_mul(m as Weight))
 			.saturating_add(T::DbWeight::get().reads(7 as Weight))
```
Comment (weichweich) [inline runtimes/spiritnet/src/weights/parachain_staking.rs:191]: we don't need the `u` variable here. maybe we can delete it?

**Pull request merged**